apps/whale-api/src/app.configuration.ts

```diff
@@ -24,3 +26,14 @@ export function AppConfiguration (): any {
     }
   }
 }
+
+export function ENV_VALIDATION_SCHEMA (): any {
+  return Joi.object({
+    NODE_ENV: Joi.string().optional(),
+    WHALE_VERSION: Joi.string().optional(),
+    WHALE_NETWORK: Joi.string().valid('mainnet', 'testnet', 'regtest', 'devnet').default('regtest'),
+    WHALE_DEFID_URL: Joi.string().optional(),
+    WHALE_DATABASE_PROVIDER: Joi.string().optional(),
+    WHALE_DATABASE_LEVEL_LOCATION: Joi.string().optional()
+  })
+}
```

Comment:
I think we can ticket a refactor declaring this like the following:
```ts
export const ENV_VALIDATION_SCHEMA = {
  return Joi.object({
    NODE_ENV: Joi.string().optional(),
    WHALE_VERSION: Joi.string().optional(),
    WHALE_NETWORK: Joi.string().valid('mainnet', 'testnet', 'regtest', 'devnet').default('regtest'),
    WHALE_DEFID_URL: Joi.string().optional(),
    WHALE_DATABASE_PROVIDER: Joi.string().optional(),
    WHALE_DATABASE_LEVEL_LOCATION: Joi.string().optional()
  })
}
```
Since this is a constant that doesn't take any parameters I don't see the need to use this as a method.

Then your implementation in https://github.com/JellyfishSDK/jellyfish/blob/c12ba51d48f297a14900b2e26bcf641a97613776/apps/whale-api/src/app.module.ts#L38 becomes:

```ts
validationSchema: ENV_VALIDATION_SCHEMA
```

Reply:
Hmm I wonder if there are any implications, maybe without the ability to do dynamic injection.